drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -105,23 +105,27 @@ int intel_adsp_hda_dma_link_in_config(const struct device *dev,
 {
 	const struct intel_adsp_hda_dma_cfg *const cfg = dev->config;
 	struct dma_block_config *blk_cfg;
+	uint32_t size = 0;
 	uint8_t *buf;
 	int res;
 
 	__ASSERT(channel < cfg->dma_channels, "Channel does not exist");
-	__ASSERT(dma_cfg->block_count == 1,
```

Comment:
Of the changes here this seems the most suspicious to me. The block_count of the dma config can only be one, there's no notion of scatter gather with HDA dma?

Reply:
Thats right, the buffer is contiguously allocated so we can calculate "single real block" size if higher block_count provided.
Thats because originally in SOF this is done by hda driver itself and hda link and gpdma can be configured from the same dai code which will divide this for blocks (but the buffer is contiguously in both cases).

**But**:  I can also do this in SOF-part If you think that it would be better to accept only a perfectly correct config at driver level, actually that might be a better approach

Reply:
I think that would be a better approach as this is really meant to provide a hardware interface. If a block stream like abstraction is needed on top that should be done on top.

I don't think this interface should be attempting to fake out that interface here. Unless, as I mentioned, it can be done with some checks here to validate that the blocks are contiguous and its documented as such, then maybe it'd be ok. I trust your judgement on this.

Reply:
Agree, it fits better to layer above driver.
Removed this commit and I will move this to SOF part